src/pudl/output/pudltabl.py

```diff
@@ -82,15 +82,14 @@ def __init__(self, pudl_engine, ds=None, freq=None, start_date=None, end_date=No
         if start_date is None:
             self.start_date = \
                 pd.to_datetime(
-                    f"{min(pc.working_years['eia923'])}-01-01")
+                    f"{min(pc.working_partitions['eia923']['years'])}-01-01")
         else:
             # Make sure it's a date... and not a string.
             self.start_date = pd.to_datetime(start_date)
 
         if end_date is None:
-            self.end_date = \
-                pd.to_datetime(
-                    f"{max(pc.working_years['eia923'])}-12-31")
+            self.end_date = pd.to_datetime(
+                pc.working_partitions['eia860m']['year_month'])
```

Comment:
It is somewhat confusing that we are pulling the date ranges from different working_partitions. The start is given by eia923, the end by eia860m. Is this guaranteed to be logically correct?

Reply:
Here we are basically trying to set the default start and end date based on the max and min EIA dates. So these are different because 923 has the oldest year rn and 860m has the newest.

To make this clear I'm thinking about incorporating a little helper function that grabs all EIA dates from `pc.working_partitions` and we can just grab the max/min.